### PR TITLE
[PAY-902] Lazily create userbanks

### PR DESCRIPTION
**Pull request by piazzatron**
### Description

Relies on https://github.com/AudiusProject/audius-protocol/pull/4795/files
Lazily creates userbanks in:
- Tipping
- Audio Page Transfers
- Rewards
- Purchase Audio
- Receive AUDIO modal

### Dragons

- This will break identity autoclaim for new users if we decide we care about that.
- Generally touches a lot of $ things

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

- Tested all core flows on web, didn't test mobile yet but it should all be shared & will add to client release checklist

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/services/audius-backend/waudio.ts [hunk 1]
```diff
@@ -1,94 +1,102 @@
...
-  audiusBackendInstance: AudiusBackend
+export const deriveUserBankPubkey = async (
+  audiusBackendInstance: AudiusBackend,
+  sourceEthAddress?: string
 ) => {
-  const audiusLibs = await audiusBackendInstance.getAudiusLibs()
-  const tokenAccount: Nullable<AccountInfo> =
-    await audiusLibs.solanaWeb3Manager.getTokenAccountInfo(addr)
-  return tokenAccount != null
+  const audiusLibs: AudiusLibs = await audiusBackendInstance.getAudiusLibs()
```
Comment (piazzatron) [inline packages/common/src/services/audius-backend/waudio.ts:11]: cc @dylanjeffers , typing libs :)

File: packages/common/src/services/audius-backend/waudio.ts [hunk 2]
```diff
@@ -1,94 +1,102 @@
...
+  return await audiusLibs.solanaWeb3Manager!.deriveUserBank(sourceEthAddress)
 }
 
-export const createUserBank = async (
-  feePayerOverride = null,
-  audiusBackendInstance: AudiusBackend
+export const deriveUserBankAddress = async (
```
Comment (piazzatron) [inline packages/common/src/services/audius-backend/waudio.ts:15]: Made this into two methods, one that returns the pubkey and one that returns a branded `SolanaWalletAddress` for extra safety

File: packages/common/src/services/wallet-client/WalletClient.ts [hunk 3]
```diff
@@ -171,7 +176,10 @@ export class WalletClient {
     }
   }
 
-  async sendWAudioTokens(address: WalletAddress, amount: BNWei): Promise<void> {
+  async sendWAudioTokens(
+    address: SolanaWalletAddress,
```
Comment (piazzatron) [inline packages/common/src/services/wallet-client/WalletClient.ts:180]: added brand

File: packages/web/src/common/store/tipping/sagas.ts [hunk 4]
```diff
@@ -230,7 +236,36 @@ function* sendTipAsync() {
...
+
+  // Create Userbanks if needed
+  const feePayerOverride = yield* select(getFeePayer)
+  if (!feePayerOverride) {
+    console.error("tippingSagas: unexpectedly couldn't get feePayerOverride")
+    return
+  }
+
+  // Gross cast here bc of broken saga types with `yield* all`
+  const [selfUserBank, recipientUserBank] = yield* all([
```
Comment (piazzatron) [inline packages/web/src/common/store/tipping/sagas.ts:249]: Technically selfUserBank should exist at this point, but being defensive

File: packages/web/src/pages/audio-rewards-page/components/ReceiveBody.tsx [hunk 5]
```diff
@@ -48,13 +57,24 @@ const useLocalStorageClickedReceiveUnderstand = (): [boolean, () => void] => {
...
+}
+
 const ReceiveBody = ({ wallet, solWallet }: ReceiveBodyProps) => {
   const useSolSPLAudio = getFeatureEnabled(
     FeatureFlags.ENABLE_SPL_AUDIO
   ) as boolean
   const [hasClickedUnderstand, onClickUnderstand] =
     useLocalStorageClickedReceiveUnderstand()
 
+  useCreateUserbankIfNeeded()
```
Comment (piazzatron) [inline packages/web/src/pages/audio-rewards-page/components/ReceiveBody.tsx:76]: Need to create userbank if advertising that a user can receive funds

**Review by rickyrombo — APPROVED** [review]
Nice! I didn't check the waudio one as thoroughly but buy audio/tipping/account ones look good?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/piazz-lazy-userbank-2

File: packages/common/src/store/solana/sagas.ts [hunk 6]
```diff
@@ -1,12 +1,11 @@
...
 
-import * as backendActions from 'common/store/backend/actions'
-const { setFeePayer } = solanaActions
+import { getContext } from '../effects'
+
+import { setFeePayer } from './slice'
 
 function* watchForFeePayer() {
   const audiusBackendInstance = yield* getContext('audiusBackendInstance')
-  yield take(backendActions.SETUP_BACKEND_SUCCEEDED)
```
Comment (rickyrombo) [inline packages/common/src/store/solana/sagas.ts:9]: do we need `waitForRead` or `waitForWrite` or something in this place?

**Review by rickyrombo — APPROVED** [review]

File: packages/common/src/store/solana/sagas.ts (hunk 6 above)
Comment (piazzatron) [reply packages/common/src/store/solana/sagas.ts:9]: nah. getRandomFeePayer has a waitForLibs already, this was unnecessary

**Review by rickyrombo — APPROVED** [review]

**Pull request merged**